easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -537,29 +538,43 @@ def include_defined_parameters(keyset):
                         else:
                             val = newval
 
-                        ebtxt.append("%s = %s" % (key, val))
+                        add_key_and_comments(key, val)
+
                         printed_keys.append(key)
                         printed = True
                 if printed:
                     ebtxt.append('')
 
+        def add_key_and_comments(key, val):
+            """ Adds key, value pair and comments (if there are any) to the dump file """
+            if key in self.comments['inline']:
```

Comment:
'outer scope' is overstating things a bit, this inner function (in a method) is just referring to a 'variable' that's available in that methods scope, which is fine

moving that function out of the method, either as private method or a stand-alone function, would be messier imho (since it'll be split of from its context)